### ISSUE TITLE
Authentication with the Passwordless Gem

### ISSUE THREAD

**Issue by holden** [avo-hq/avo#1628]
### Context

I'm moving away from Devise on a current project and trying out the [passwordless](https://github.com/mikker/passwordless) gem.  You set up the `current_user` helper yourself like below however it doesn't appear enough for Avo to authenticate as I get a no method found error `undefined method `current_user' `

How can I define this so that I can use it with Avo?

Thanks

```
class ApplicationController < ActionController::Base
  include Passwordless::ControllerHelpers # <-- This!

  # ...

  helper_method :current_user

  private

  def current_user
    @current_user ||= authenticate_by_session(User)
  end

  def require_user!
    return if current_user
    redirect_to root_path, flash: { error: 'You are not worthy!' }
  end
end
```

### System configuration
**Avo version**: 2.27.1

**Rails version**: 7.0.4

**Ruby version**: 3.1.2

**License type**

 - [x] Community
 - [ ] Pro

**Are you using Avo monkey patches or overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

**Comment by adrianthedev** [comment]
Avo does not inherit from that `ApplicationController`, but has it's own. That's by design.

Follow [this](https://avohq.io/blog/safely-extend-a-ruby-on-rails-controller) article to add the `current_user` method to Avo's `ApplicationController`.

The dummy app does something similar with setting the `Current.user` object [here](https://github.com/avo-hq/avo/blob/main/spec/dummy/config/initializers/avo.rb#L144) and [here](https://github.com/avo-hq/avo/blob/main/spec/dummy/app/avo/concerns/application_controller_extensions.rb).

## TL;DR

Move those methods in a concern and load that concern in Avo's ApplicationController.

**Comment by adrianthedev** [comment]
Hey @holden. Did you manage to get this working?

**Comment by holden** [comment]
Yes! It worked beautifully, although it didn't like the `helper_method :current_user` which passwordless recommended but otherwise as a public method works fine.

```
module AvoCurrentUser
  extend ActiveSupport::Concern
  include Passwordless::ControllerHelpers

  def current_user
    @current_user ||= authenticate_by_session(User)
  end

end
```

Thanks again!
